Adds a Getting Started with Java tutorial and Java sample code to other tutorials. (#1058)
XRPLF/xrpl-dev-portal: content/tutorials/get-started/get-started-using-java.md

* The basic building blocks of XRP Ledger-based applications.
* How to connect to the XRP Ledger using `xrpl4j`.
* How to generate a wallet on the [Testnet](xrp-testnet-faucet.html) using `xrpl4j`.
* How to use the `xrpl4j` library to look up information about an account on the XRP Ledger.
* How to put these steps together to create a simple Java app.

## Requirements

* The `xrpl4j` library supports Java 1.8 and later.
* A project management tool such as [Maven](https://maven.apache.org/) or [Gradle](https://gradle.org/).

## Installation

The [`xrpl4j` library](https://github.com/XRPLF/xrpl4j) is available on [Maven Central](https://search.maven.org/artifact/org.xrpl/xrpl4j-parent).
`xrpl4j` is split into multiple artifacts, which can be imported as needed.

In this tutorial, you will need the [xrpl4j-client](https://javadoc.io/doc/org.xrpl/xrpl4j-client/latest/index.html), 
[xrpl4j-address-codec](https://javadoc.io/doc/org.xrpl/xrpl4j-address-codec/latest/index.html), 
[xrpl4j-keypairs](https://javadoc.io/doc/org.xrpl/xrpl4j-keypairs/latest/index.html), and 
[xrpl4j-model](https://javadoc.io/doc/org.xrpl/xrpl4j-model/latest/index.html)  modules. 

Install with Maven:

> ```xml

Reviewer: 1. Please leave an empty line before fenced code samples. (Sometimes the Markdown parser does weird things if one is missing, though it seems to work OK in this case right now.)

2. I suggest making the introduction more specific:

```suggestion
To install with Maven, add the following to your project's `pom.xml` file and then run `mvn install`:

```xml
```

Reviewer: This would also be a good place to link to the code sample repo.

<dependencies>
    <dependency>
      <groupId>org.xrpl</groupId>
      <artifactId>xrpl4j-client</artifactId>
      <version>2.0.0</version>
    </dependency>
    <dependency>
      <groupId>org.xrpl</groupId>
      <artifactId>xrpl4j-address-codec</artifactId>
      <version>2.0.0</version>
    </dependency>
    <dependency>
      <groupId>org.xrpl</groupId>
      <artifactId>xrpl4j-keypairs</artifactId>
      <version>2.0.0</version>
    </dependency>
    <dependency>
      <groupId>org.xrpl</groupId>
      <artifactId>xrpl4j-model</artifactId>
      <version>2.0.0</version>
    </dependency>

> </dependencies>

Reviewer: If you go with my suggestion of providing a default config file in the code samples repo, you can use `{{include_code()}}` to pull in the relevant portion of the file here so it doesn't need to be maintained in two places.

AccountInfoResult{
    status=success, 
    accountData=AccountRootObject{
        ledgerEntryType=ACCOUNT_ROOT, 
        account=rBXHGshqXu3Smy9FUsQTmo49bGpQUQEm3X, 
        balance=1000000000, 
        flags=0, 
        ownerCount=0, 
        previousTransactionId=0000000000000000000000000000000000000000000000000000000000000000, 
        previousTransactionLedgerSequence=0, 
        sequence=17178149, 
        signerLists=[], 
        index=0DC1B13C73A7F3D2D82446526D0C5D08E88F89BA442D54291117F1A08E447685
    }, 
    ledgerCurrentIndex=17178149, 
    validated=false
}
```

#### Interpreting the response

The response fields contained in `AccountInfoResult` that you want to inspect in most cases are:

* `accountData.sequence` — This is the sequence number of the next valid transaction for the account. You need to specify the sequence number when you prepare transactions.

* `accountData.balance` — This is the account's balance of XRP, in drops. You can use this to confirm that you have enough XRP to send (if you're making a payment) and to meet the [current transaction cost](transaction-cost.html#current-transaction-cost) for a given transaction.

> * `validated` — Indicates whether the returned data is from a [validated ledger](ledgers.html#open-closed-and-validated-ledgers). When sending transactions, it's important to ensure that the results are in a [final](finality-of-results.html) state in a validated ledger before further processing the transaction. For more information about best practices for transaction processing, see [Reliable Transaction Submission](reliable-transaction-submission.html).

Reviewer: > When sending transactions, it's important to ensure that the results are in a [final](finality-of-results.html) state in a validated ledger

Do you mean when _receiving_ transactions here? Or when processing the outcome of a transaction you sent?

Also, "in a final state in a validated ledger" is not quite right. By definition, being in a validated ledger is a final state, but it's not the only way things can be in a final state. Perhaps the following instead:

```suggestion
* `validated` — Indicates whether the returned data is from a [validated ledger](ledgers.html#open-closed-and-validated-ledgers). When inspecting transactions, it's important to confirm that [the results are final](finality-of-results.html) before further processing the transaction. If `validated` is `true` then you know for sure the results won't change. For more information about best practices for transaction processing, see [Reliable Transaction Submission](reliable-transaction-submission.html).
```

Author: Totally agree. This was copy pasted from the python Getting Started tutorial, so we should probably update that as well.

In this tutorial, you will need the [xrpl4j-client](https://javadoc.io/doc/org.xrpl/xrpl4j-client/latest/index.html), 
[xrpl4j-address-codec](https://javadoc.io/doc/org.xrpl/xrpl4j-address-codec/latest/index.html), 
[xrpl4j-keypairs](https://javadoc.io/doc/org.xrpl/xrpl4j-keypairs/latest/index.html), and 
[xrpl4j-model](https://javadoc.io/doc/org.xrpl/xrpl4j-model/latest/index.html)  modules. 

To install with Maven, add the following to your project's `pom.xml` file and then run `mvn install`:

```xml
<dependencies>
    <dependency>
      <groupId>org.xrpl</groupId>
      <artifactId>xrpl4j-client</artifactId>
      <version>2.0.0</version>
    </dependency>
    <dependency>
      <groupId>org.xrpl</groupId>
      <artifactId>xrpl4j-address-codec</artifactId>
      <version>2.0.0</version>
    </dependency>
    <dependency>
      <groupId>org.xrpl</groupId>
      <artifactId>xrpl4j-keypairs</artifactId>
      <version>2.0.0</version>
    </dependency>
    <dependency>
      <groupId>org.xrpl</groupId>
      <artifactId>xrpl4j-model</artifactId>
      <version>2.0.0</version>
    </dependency>
</dependencies>
```

> Check out the [xrpl4j sample project](https://github.com/XRPLF/xrpl4j-sample) for a full maven project containing the code from this tutorial.

Reviewer: ```suggestion
Check out the [xrpl4j sample project](https://github.com/XRPLF/xrpl4j-sample) for a full Maven project containing the code from this tutorial.
```

Here are the basic steps you'll need to cover for almost any XRP Ledger project:

1. [Connect to the XRP Ledger.](#1-connect-to-the-xrp-ledger)
1. [Generate a wallet.](#2-generate-wallet)
1. [Query the XRP Ledger.](#3-query-the-xrp-ledger)

### {{n.next()}}. Connect to the XRP Ledger

To make queries and submit transactions, you need to establish a connection to the XRP Ledger. To do this with `xrpl4j`, 
you can use an [`XrplClient`](https://javadoc.io/doc/org.xrpl/xrpl4j-client/latest/org/xrpl/xrpl4j/client/XrplClient.html):

{{ include_code("_code-samples/xrpl4j/GetAccountInfo.java", start_with="// Construct a network client", end_before="// Create a Wallet using a WalletFactory", language="java") }}

#### Connect to the production XRP Ledger

The sample code in the previous section shows you how to connect to the Testnet, which is one of the available [parallel networks](parallel-networks.html). When you're ready to integrate with the production XRP Ledger, you'll need to connect to the Mainnet. You can do that in two ways:

> * By [installing the core server](install-rippled.html) (`rippled`) and running a node yourself (the core server connects to the Mainnet by default and you can [change the configuration to use an altnet](connect-your-rippled-to-the-xrp-test-net.html) ). [There are good reasons to run your own core server](the-rippled-server.html#reasons-to-run-your-own-server). If you run your own server, you can connect to it like so:

Reviewer: Another sentence we should fix in this _and_ the Python one. ;)

```suggestion
* By [installing the core server](install-rippled.html) (`rippled`) and running a node yourself. The core server connects to the Mainnet by default, but you can [change the configuration to use Testnet or Devnet](connect-your-rippled-to-the-xrp-test-net.html). [There are good reasons to run your own core server](the-rippled-server.html#reasons-to-run-your-own-server). If you run your own server, you can connect to it like so:
```
